rust/datafusion/src/optimizer/utils.rs

```diff
@@ -370,6 +385,27 @@ pub fn rewrite_expression(expr: &Expr, expressions: &Vec<Expr>) -> Result<Expr>
             asc: asc.clone(),
             nulls_first: nulls_first.clone(),
         }),
+        Expr::Between { negated, .. } => {
+            let expr = Expr::BinaryExpr {
+                left: Box::new(Expr::BinaryExpr {
+                    left: Box::new(expressions[0].clone()),
+                    op: Operator::GtEq,
+                    right: Box::new(expressions[1].clone()),
+                }),
+                op: Operator::And,
+                right: Box::new(Expr::BinaryExpr {
+                    left: Box::new(expressions[0].clone()),
+                    op: Operator::LtEq,
+                    right: Box::new(expressions[2].clone()),
+                }),
+            };
+
+            if *negated {
+                Ok(Expr::Not(Box::new(expr)))
```

Comment:
nit: it would probably be more efficient to create the `>=` and `<=` with the values reversed rather than wrapping in a `NOT` since this adds one more operation and array creation. This is the sort of thing that could be fixed in the opimizer phase too.

Reply:
On second thoughts, we should leave the logical plan as it is so we don't lose the semantics. We can optimize this in the physical plan in the future.

Reply:
so would you like to make changes? I'm happy to implement whatever decision you make.

Reply:
I think it is indeed better to keep it as is, and do the optimization in a optimization rule

Reply:
So no changes needed here